File: backend/src/main/java/com/example/backend/models/Product.java
package com.example.backend.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.util.List;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "products")
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_product")
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, name = "image_url")
    private String imageUrl;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal price;

    @Column(nullable = false)
    private Integer stock;

    private boolean status = true; // toods inician en true, si c acaba pasa false

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_category")
    private Category category;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
            name = "product_sizes", // nombre de la tabla intermedia en la bd
            joinColumns = @JoinColumn(name = "id_product"),
            inverseJoinColumns = @JoinColumn(name = "id_size")
    )
    private List<Size> sizes;
}

File: frontend/src/types/product.ts
export interface ProductDetail {
    id: number;
    name: string;
    price: number;
    stock: number;
    imageUrl: string;
    status: boolean;
    categoryName: string;
    description: string;
    sizes: ProductSize[];
    relatedProducts: RelatedProduct[];
}

export interface ProductSize {
    id: number;
    name: string;
    dimension: string;
    price: number;
}

export interface RelatedProduct {
    id: number;
    name: string;
    price: number;
    imageUrl: string;
}
